jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialManager.kt

```diff
@@ -6,75 +6,151 @@ package software.aws.toolkits.jetbrains.core.credentials
 import com.intellij.openapi.Disposable
 import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.extensions.ExtensionPointName
 import com.intellij.openapi.util.Disposer
 import com.intellij.openapi.util.SimpleModificationTracker
 import com.intellij.util.messages.MessageBus
 import com.intellij.util.messages.Topic
 import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
 import software.amazon.awssdk.auth.credentials.AwsSessionCredentials
 import software.aws.toolkits.core.credentials.CredentialProviderNotFound
-import software.aws.toolkits.core.credentials.DefaultToolkitCredentialsProviderManager
 import software.aws.toolkits.core.credentials.ToolkitCredentialsChangeListener
+import software.aws.toolkits.core.credentials.ToolkitCredentialsIdentifier
 import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.core.AwsSdkClient
+import java.util.concurrent.ConcurrentHashMap
+import javax.security.auth.login.CredentialNotFoundException
+
+data class CredentialIdentifierChange(
```

Comment:
The name of this class really confused me, potentially something like `CredentialsChangeEvent` would be more clear

Reply:
Agree, will change to your suggestion